src/vs/workbench/services/extensions/common/extensions.ts

```diff
@@ -443,7 +443,12 @@ export interface IExtensionService {
 	/**
 	 * Fired when the available extensions change (i.e. when extensions are added or removed).
 	 */
-	onDidChangeExtensions: Event<void>;
+	onDidChangeExtensions: Event<{ readonly added: readonly IExtensionDescription[]; readonly removed: readonly IExtensionDescription[] }>;
+
+	/**
+	 * All registered extensions. Can be empty.
```

Comment:
Just a small suggestion to improve the JSDoc / to explain that:
* the list will always be empty initially during workbench startup
* to react to the initial extension scanning / the initialization of extensions, a consumer can use `onDidRegisterExtensions` or `whenInstalledExtensionsRegistered`.
* the list will change as extensions get installed, uninstalled, enabled, disabled, so the event `onDidChangeExtensions` can be used to observe all changes.